controllers/replication.storage/volumereplication_controller.go

```diff
@@ -203,17 +223,19 @@ func (r *VolumeReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 
 			return reconcile.Result{}, err
 		}
+		switch instance.Spec.DataSource.Kind {
```

Comment:
All the Volume group replication related finalizers will be handled by VGR_contorller, so I don't think we need a switch case here as such, we can simply use an if condition for the pvcDataSource.
Or, do we plan to add any further finalizers here for vgr?

Reply:
@Nikhil-Ladha Each PVC needs to be handled by the VGR now, we cannot handle PVC in VR as VR will not get any trigger at PVC level for group replication, based on the dataSource this will set the finalizers and owner on the VGR

Reply:
Ack